sklearn/cluster/dbscan_.py

```diff
@@ -111,7 +114,14 @@ def dbscan(X, eps=0.5, min_samples=5, metric='minkowski',
         neighbors_model.fit(X)
         neighborhoods = neighbors_model.radius_neighbors(X, eps,
                                                          return_distance=False)
-        neighborhoods = np.array(neighborhoods)
+
+    # Hack to make dtype=object, ndim=1 array when all neighborhoods
```

Comment:
Good catch. All neighbors models are now consistently return `dtype=object` arrays for radius queries in master. I removed this hack in my branch and improved the test I added to check for `algorithm` in `['brute', 'kd_tree', 'ball_tree']`.